File: cardio-guard-backend/src/main/java/com/cardioguard/entity/EcgAnnotation.java
package com.cardioguard.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * ECG心电图标注实体类
 */
@Data
@TableName("ecg_annotation")
public class EcgAnnotation implements Serializable {
    
    private static final long serialVersionUID = 1L;
    
    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;
    
    /**
     * 用户ID
     */
    private Long userId;
    
    /**
     * 关联的ECG分析结果ID
     */
    private Long analysisResultId;
    
    /**
     * 标注点的时间戳(毫秒,相对于记录开始)
     */
    private Long timestampMs;
    
    /**
     * 标注类型
     * P_WAVE - P波
     * QRS_COMPLEX - QRS波群
     * T_WAVE - T波
     * ST_SEGMENT - ST段
     * ABNORMAL_POINT - 异常点
     * OTHER - 其他
     */
    private String annotationType;
    
    /**
     * 标注标签/描述
     */
    private String label;
    
    /**
     * 标注置信度(0-1, AI自动标注时)
     */
    private Double confidence;
    
    /**
     * 是否AI自动生成(0-手动, 1-AI)
     */
    private Integer isAiGenerated;
    
    /**
     * 创建者ID(医生或用户)
     */
    private Long createdBy;
    
    /**
     * 创建时间
     */
    private LocalDateTime createdAt;
    
    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
}

File: cardio-guard-frontend/src/api/ecg-annotation.ts
import axios from 'axios'

const BASE_URL = '/api/ecg/annotation'

/**
 * ECG标注类型枚举
 */
export enum AnnotationType {
  P_WAVE = 'P_WAVE',
  QRS_COMPLEX = 'QRS_COMPLEX',
  T_WAVE = 'T_WAVE',
  ST_SEGMENT = 'ST_SEGMENT',
  ABNORMAL_POINT = 'ABNORMAL_POINT',
  OTHER = 'OTHER'
}

/**
 * ECG标注接口
 */
export interface EcgAnnotation {
  id?: number
  userId: number
  analysisResultId: number
  timestampMs: number
  annotationType: AnnotationType | string
  label?: string
  confidence?: number
  isAiGenerated?: number
  createdBy: number
  createdAt?: string
  updatedAt?: string
}

/**
 * 标注统计信息
 */
export interface AnnotationStatistics {
  totalCount: number
  typeStatistics: Record<string, number>
  aiGeneratedCount: number
  manualCount: number
}

/**
 * 创建标注
 */
export function createAnnotation(annotation: EcgAnnotation) {
  return axios.post(BASE_URL, annotation)
}

/**
 * 批量创建标注
 */
export function batchCreateAnnotations(annotations: EcgAnnotation[]) {
  return axios.post(`${BASE_URL}/batch`, annotations)
}

/**
 * 更新标注
 */
export function updateAnnotation(annotationId: number, annotation: Partial<EcgAnnotation>) {
  return axios.put(`${BASE_URL}/${annotationId}`, annotation)
}

/**
 * 删除标注
 */
export function deleteAnnotation(annotationId: number, userId: number) {
  return axios.delete(`${BASE_URL}/${annotationId}`, {
    params: { userId }
  })
}

/**
 * 查询指定分析结果的标注
 */
export function getAnnotationsByResultId(analysisResultId: number) {
  return axios.get(`${BASE_URL}/result/${analysisResultId}`)
}

/**
 * 查询用户标注历史
 */
export function getUserAnnotationHistory(userId: number, page: number = 1, size: number = 20) {
  return axios.get(`${BASE_URL}/history`, {
    params: { userId, page, size }
  })
}

/**
 * AI自动标注
 */
export function aiAutoAnnotate(analysisResultId: number, userId: number) {
  return axios.post(`${BASE_URL}/ai-annotate/${analysisResultId}`, null, {
    params: { userId }
  })
}

/**
 * 查询用户标注统计
 */
export function getUserAnnotationStatistics(userId: number) {
  return axios.get(`${BASE_URL}/statistics`, {
    params: { userId }
  })
}
